File: src/main/java/grojdg/harderstats/callbacks/PlayerManagerCallbacks.java
package grojdg.harderstats.callbacks;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.util.ActionResult;

public class PlayerManagerCallbacks {
    public static final Event<AfterPlayerConnect> AFTER_PLAYER_CONNECT = EventFactory.createArrayBacked(AfterPlayerConnect.class,
            (listeners) -> (player) -> {
                for (AfterPlayerConnect listener : listeners) {
                    listener.interact(player);
                }
            });

    public interface AfterPlayerConnect {
        void interact(ServerPlayerEntity player);
    }
}

File: src/main/kotlin/grojdg/harderstats/callbacks/PlayerManagerCallbacks.kt
package grojdg.harderstats.callbacks

import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.server.network.ServerPlayerEntity

fun interface AfterPlayerConnect {
    fun interact(player: ServerPlayerEntity)

    companion object {
        val event = EventFactory.createArrayBacked(AfterPlayerConnect::class.java
        ) { listeners ->
            AfterPlayerConnect { player ->
                for (listener in listeners) {
                    listener.interact(player)
                }
            }
        }
    }
}